bio/slow5/slow5.go

```diff
@@ -473,3 +475,50 @@ func (read *Read) WriteTo(w io.Writer) (int64, error) {
 	}
 	return written, nil
 }
+
+/******************************************************************************
+Aug 15, 2023
+
+StreamVByte (svb) compression of raw signal strength is used to decrease the
+overall size of records in blow5 files. In my tests using a SQLite database
+that contained both slow5 and fastq files, switching from raw signal TEXT to
+svb compressed BLOBs brought the total database size from 12GB to 7.1GB, a
+~40% reduction in size.
+
+This is the primary method that can be used to decrease the size of storing
+slow5 records in alternative datastores to blow5 (SQL databases, etc).
+
+svb is an integer compression algorithm, which are specialized in compressing
+integer arrays, which perfectly fits raw signal strength data. When converting
+from slow5 to blow5, files are additionally compressed with zstd or zlib on top
+of raw signal compression with svb. Still, svb compression is where most the
+data size saving comes from.
+
+Stream VByte paper: https://doi.org/10.48550/arXiv.1709.08990
+
+Keoni
+
+******************************************************************************/
+
+// SvbCompressRawSignal takes a read and converts its raw signal field to two
+// arrays: a mask array and a data array. Both are needed for decompression.
+func SvbCompressRawSignal(rawSignal []int16) (mask, data []byte) {
```

Comment:
I think it would be helpful to define a `CompressedRawSignal` struct to bundle `mask`, `data`, and `len(rawSignal)` as AFAICT, these data do not make sense independent of each other (? not sure, but this is what I'm guessing).

Reply:
I'm not sure actually. For additional context, all 3 (mask, data, len(rawSignal)) would actually be stored somewhere, so you would get them out separately (like in an SQL database, you'd have a `mask` column and a `data` column). At the same time, I could imagine if you want to store in a JSON format or a document store, it'd make way more sense to have all the numbers together, so you could just pass in a marshaled struct directly from the JSON without having to access the data... although you could just have that struct have a method to call Svb compress/decompress

Reply:
Because of the IO nature of this, I'm just gonna keep it as a function for now.